src/main/java/net/neoforged/neoforge/common/util/NeoForgeExtraCodecs.java

```diff
@@ -315,4 +317,44 @@ public String toString() {
             return "XorMapCodec[" + first + ", " + second + "]";
         }
     }
+
+    /**
+     * Creates a codec for an {@linkplain Codec#unboundedMap(Codec, Codec) unbounded map} whose underlying representation is a list of maps, with the given names for each key-element entry. Each key-element entry is encoded as a map with the given key and element names respectively.
+     * <p>
+     * This is useful for maps where the key does not encode to a string, which causes errors when trying to serialize to a format that requires maps to have string keys (such as JSON and NBT).
+     *
+     * @param keyName      the name of the key in the encoded map for each key-element entry
+     * @param keyCodec     codec for the key
+     * @param elementName  the name of the element in the encoded map for each key-element entry
+     * @param elementCodec codec for the element
+     * @param <K>          the key type
+     * @param <V>          the element type
+     * @return a codec for an unbounded map
+     */
+    public static <K, V> Codec<Map<K, V>> unboundedMapAsList(String keyName, Codec<K> keyCodec, String elementName, Codec<V> elementCodec) {
+        Codec<Pair<K, V>> pairCodec = RecordCodecBuilder.create(
+                instance -> instance.group(
+                        keyCodec.fieldOf(keyName).forGetter(Pair::getFirst),
+                        elementCodec.fieldOf(elementName).forGetter(Pair::getSecond)).apply(instance, Pair::of));
+        return Codec.list(pairCodec)
+                .xmap(
+                        entries -> entries.stream().collect(Pair.toMap()),
+                        map -> map.entrySet().stream().map(entry -> Pair.of(entry.getKey(), entry.getValue())).toList());
+    }
+
+    /**
+     * Creates a codec for an {@linkplain Codec#unboundedMap(Codec, Codec) unbounded map} whose underlying representation is a list of maps, with names of {@code "key"} and {@code "value"} for each key-element entry.
+     * <p>
+     * This is useful for maps where the key does not encode to a string, which causes errors when trying to serialize to a format that requires maps to have string keys (such as JSON and NBT).
+     *
+     * @param keyCodec     codec for the key
+     * @param elementCodec codec for the element
+     * @param <K>          the key type
+     * @param <V>          the element type
+     * @return a codec for an unbounded map
+     * @see #unboundedMapAsList(String, Codec, String, Codec)
+     */
+    public static <K, V> Codec<Map<K, V>> unboundedMapAsList(Codec<K> keyCodec, Codec<V> elementCodec) {
+        return unboundedMapAsList("key", keyCodec, "element", elementCodec);
```

Comment:
I would use `key` and `value`. Alternatively, I would simply not add this method and prefer if the names are always explicit.

Reply:
I used `"key"` and `"element"` because `Codec#unboundedMap` uses those names for its two parameter codecs: `keyCodec` and `elementCodec`.

As for the existence of this method, I added it for convenience. I can just remove it if that's wanted, since even in this PR I used explicit names.

Reply:
I would remove it to encourage the use of descriptive names.